app/assets/javascripts/master_detail.js

```diff
@@ -89,48 +89,62 @@ function clickBack(btn, level) {
   });
 }
 
-function reset_grid(selector, level){
+function reset_grid(selector, level, factory_reset = false){
   var $grid = $(selector);
   var dataGrid;
+  var selectedRowKeys;
+  var preserve_selected_rows = false;
```

Comment:
`var preserve_selected_rows = $grid.data("master_detail-preserve_selected_rows") ?? false;`

Reply:
Please add comments for each of these properties. Verbose commenting so that when we refactor to use different methodologies to sort out the state persistence, we do not erroneously remove code.